cli/src/runner.rs

```diff
@@ -172,61 +178,65 @@ pub async fn extract_failed_tests(
 }
 
 pub async fn run_quarantine(
-    run_result: &RunResult,
+    run_result: RunResult,
     api_address: &str,
     token: &str,
     org_url_slug: &str,
     repo: &BundleRepo,
     delay: std::iter::Take<ExponentialBackoff>,
 ) -> anyhow::Result<QuarantineRunResult> {
-    let quarantine_results = if run_result.failures.is_empty() {
-        QuarantineBulkTestStatus {
-            group_is_quarantined: false,
-            quarantine_results: Vec::new(),
-        }
-    } else {
+    let quarantine_config: QuarantineConfig = if !run_result.failures.is_empty() {
         log::info!("Quarantining failed tests");
         let result = Retry::spawn(delay, || {
-            get_quarantine_bulk_test_status(
-                api_address,
-                token,
-                org_url_slug,
-                &repo.repo,
-                &run_result.failures,
-            )
+            get_quarantining_config(api_address, token, org_url_slug, &repo.repo)
         })
         .await;
 
-        result.unwrap_or_else(|e| {
-            log::error!("Failed to get quarantine results: {:?}", e);
-            QuarantineBulkTestStatus {
-                group_is_quarantined: false,
-                quarantine_results: Vec::new(),
-            }
-        })
+        if let Err(ref err) = result {
+            log::error!("Failed to get quarantine results: {:?}", err);
+        }
+        result.unwrap_or_default()
+    } else {
+        log::debug!("No failed tests to quarantine");
+        QuarantineConfig::default()
     };
 
-    // Print out failed tests and their quarantine status
-    let quarantined_names: HashSet<_> = quarantine_results
-        .quarantine_results
-        .iter()
-        .map(|qr| &qr.name)
+    // quarantine the failed tests
+    let mut quarantine_results = QuarantineBulkTestStatus::default();
+    let quarantined: HashSet<_> = quarantine_config.quarantined_tests.iter().collect();
```

Comment:
if you define `quarantined_tests` as a `HashSet` you won't have to convert it to one here

Reply:
ah shoot forgot abut that